src/components/AdobeSignDialog.vue

```diff
@@ -0,0 +1,156 @@
+            placeholder="Email"
+            type="email"
+            enabled
+          />
+        </div>
+        <div class="flex justify-content-center">
+          <PvButton
+            label="Done"
+            class="no-underline bg-primary mt-2 border-none border-round p-3 w-5 text-white hover:bg-red-900"
+            @click="createConsent"
+          />
+        </div>
+      </div>
+      <div v-if="hasSigningUrl" class="bg-white border-1 border-round w-full text-center">
+        <div class="flex flex-row">
+          <div class="justify-content-center flex">
+            <img src="../assets/signDoc2.webp" style="width: 40vh; padding: 0px" />
+          </div>
+          <div class="flex flex-column justify-content-center p-5">
+            <h3>
+              Thank you! An email has been sent to <span class="font-bold">{{ signerEmail }}</span> with further
+              instructions to sign the consent form. <br />
+              <br />
+              Please go to your inbox and complete the signing process.
+            </h3>
+            <h3>Alternatively, you can click the link below to go directly to the signing page:</h3>
+            <h4>
+              <a :href="adobeUrl" target="blank"> Sign the form now </a>
+            </h4>
+          </div>
+        </div>
+      </div>
+      <div v-if="!hasSigningUrl && docCreated">
+        <div class="flex flex-row">
+          <div class="justify-content-center flex">
+            <img src="../assets/loadinglion.webp" style="width: 40vh; padding: 0px" />
+          </div>
+          <div class="flex flex-column justify-content-center p-5">
+            <h3>We are creating your agreement document.</h3>
+            <h4>Please wait a few seconds</h4>
+          </div>
+        </div>
+      </div>
+    </template>
+  </PvDialog>
+</template>
+
+<script setup>
+import { ref, onMounted } from 'vue';
+import { useToast } from 'primevue/usetoast';
+import { useAuthStore } from '@/store/auth';
+import { storeToRefs } from 'pinia';
+
+const authStore = useAuthStore();
+const { roarfirekit } = storeToRefs(authStore);
+const initialized = ref(false);
+const agreementId = ref(null);
+const docStatus = ref(null);
+const adobeUrl = ref(null);
+const hasSigningUrl = ref(false);
+let unsubscribe;
+
+const init = () => {
+  if (unsubscribe) unsubscribe();
+  initialized.value = true;
+};
+
+unsubscribe = authStore.$subscribe(async (mutation, state) => {
+  if (state.roarfirekit.restConfig) init();
+});
+
+onMounted(() => {
+  if (roarfirekit.value.restConfig) init();
+});
+
+const signerEmail = ref('');
+const docCreated = ref(false);
+const isVisible = ref(false);
+const isAdult = ref(false);
+const emit = defineEmits(['consent-selected']);
+
+const toast = useToast();
+
+const props = defineProps({
+  isAdobe: { type: Boolean, required: true, default: false },
+  isAdult: { type: Boolean, required: true, default: false },
+  parentEmail: { type: String, required: false },
```

Comment:
```suggestion
  parentEmail: { type: String, required: false, default: null },
```
My IDE seems to think that this needs a default value.

Reply:
Thanks for catching this!